### PR TITLE
Add talent challenge program

### PR DESCRIPTION
**Pull request by Soline324**
Talent Challenge program is a platform that offers a structured remote mentoring program for college students successfully completing talent plan courses or having contributed to TiDB projects. Open source projects related to TiDB ecosystems will be selected together with professional mentors. Students successfully completing talent plan courses or having contributed to TiDB projects can apply for the projects, one at a time, and finish the projects with the help of the mentors in a given time. Mentees gain real-world experience by working closely with mentors. Students who successfully complete the projects will receive stipends and better job opportunities.

We are now launching talent challenge program season 1, here in this pull request descriptions of what talent challenge program is, how to participate in, timeline of season 1 are added.

@AndreMouche PTAL~

### CODE REVIEW

File: talent-challenge-program/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+*Note:*
+
+- *Projects to be selected must have one mentor at least.*
+- *Mentors should be prepared to donate their time to help mentees with the project.*
+
+### Mentees
+
+*Note:*
+
+- Only **students who have completed talent plan courses and having contributed to TiDB projects** can apply for Talent Challenge program.
```
Comment (Soline324) [inline talent-challenge-program/README.md:32]:
```suggestion
- Only **students who have completed talent plan courses or having contributed to TiDB projects** can apply for Talent Challenge program. 
```

File: talent-challenge-program/MENTEE_APPLY_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Mentee Apply Template
+
+Mail to: talent-plan@tidb.com
```
Comment (winkyao) [inline talent-challenge-program/MENTEE_APPLY_TEMPLATE.md:3]: talent-plan@tidb.io?

Comment (Soline324) [inline talent-challenge-program/MENTEE_APPLY_TEMPLATE.md:3]:
```suggestion
Mail to: talent-plan@tidb.io
```

File: talent-challenge-program/selected-projects.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+* Upstream Issue or RFC (URL): https://github.com/pingcap/tidb/issues/6840
+
+##### Support Global Index for Partition Table
+
+* Description: Currently TiDB only supports local partitioned indexes that compatible with mysql，but other database(e.g. Oracle) also supports global partitioned indexes
+* Recommended Skills: Go, DDL, SQL
+* Mentor(s): @tiancaiamao
+* Upstream Issue or RFC (URL): https://github.com/pingcap/tidb/issues/18032
+
+#### PD
```
Comment (Soline324) [inline talent-challenge-program/selected-projects.md:81]:
```suggestion
```

File: talent-challenge-program/selected-projects.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+* Recommended Skills: Go
+* Mentor(s): @wjhuang2016
+* Upstream Issue or RFC (URL): https://github.com/pingcap/tidb/issues/17596
+
+#### TiKV
+
+##### SQL Statement Level Statistics
+
+* Description: To make it possible to know what statement is the root cause of hot regions, we need to add local statistics to TiKV that counts top K hot statements and then reports them to PD.
+* Recommended Skills: Rust, Go
```
Comment (Soline324) [inline talent-challenge-program/selected-projects.md:102]:
```suggestion
* Recommended Skills: Rust, Go,  Algorithm / Data Structure
```

File: talent-challenge-program/selected-projects.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+##### HBase protocol support on Zetta
+
+* Description: Zetta is designed as a BigTable implementation on top of TiKV to support structured schemaless data model. In order to elaborate existing HBase ecosystem, we plan to implement the HBase protocol adaptor for Zetta. To achieve this, we need to:
+	1. Implement a mock ZooKeeper to eliminate external dependency
+	2. Implement RPC protocol of HBase RegionServer
+	3. Implement HBase functionality on top of Zetta
+* Recommended Skills: Go, Java, HBase RPC
+* Mentor(s): pseudocodes, baiyuqing
+* Upstream Issue or RFC (URL): WIP
```
Comment (Soline324) [inline talent-challenge-program/selected-projects.md:123]:
```suggestion
* Upstream Issue or RFC (URL): https://github.com/zhihu/zetta/issues/2
```

File: talent-challenge-program/selected-projects.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
+## Selected Projects in Season 1
+
+Selected Projects in Season 1 are listed below. Mentee candidates who want to participate in a certain project please refer to [Mentee Selection Process](README.md#mentees)
+
```
Comment (Soline324) [inline talent-challenge-program/selected-projects.md:4]:
```suggestion
The difficulty of the project is divided into three levels: Very hard, hard, and medium, the corresponding pre-tax bonus is RMB 10000 to a very hard project, RMB 8000 to a hard project and RMB 5000 to a medium project respectively. 
```

File: talent-challenge-program/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+*Note:*
+
+- *Projects to be selected must have one mentor at least.*
+- *Mentors should be prepared to donate their time to help mentees with the project.*
+
+### Mentees
+
+*Note:*
+
+- Only **students who have completed talent plan courses or having contributed to TiDB projects** can apply for Talent Challenge program.
```
Comment (gingerkidney) [inline talent-challenge-program/README.md:32]: Can contributors apply for the project directly?

Comment (gingerkidney) [reply talent-challenge-program/README.md:32]: the rest LGTM

File: talent-challenge-program/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Talent Challenge Program
+
+Talent Challenge program is a platform that offers a structured remote mentoring program for **college students** successfully completing talent plan courses or having contributed to TiDB projects. Open source projects related to TiDB ecosystems will be selected together with professional mentors. Students **successfully completing talent plan courses or having contributed to TiDB projects** can apply for the projects, one at a time, and finish the projects with the help of the mentors in a given time. Mentees gain real-world experience by working closely with mentors. Students who successfully complete the projects will receive **stipends and better job opportunities**.
```
Comment (Soline324) [inline talent-challenge-program/README.md:3]:
```suggestion
Talent Challenge program is a platform that offers a structured remote mentoring program for **college students** successfully completing talent plan courses or having contributed to TiDB projects. Open source projects related to TiDB ecosystems will be selected together with professional mentors. Students **successfully completing talent plan courses or having contributed to TiDB projects** can apply for the projects, one at a time, and finish the projects with the help of the mentors in a given time. Mentees gain real-world experience by working closely with mentors. Students who successfully complete the projects will receive **bonus and better job opportunities**.
```

File: talent-challenge-program/README.md (hunk 7 above)
Comment (Soline324) [reply talent-challenge-program/README.md:32]: Contributors who are college students at school can apply for the projects

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by sykp241095** [comment]
LGTM

File: talent-challenge-program/project-ideas.md [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+### Proposed Project ideas
+
+#### BR
+
+##### BR HTTP Storage
+- Description: Support HTTP(S) server as source and destination for BR, and allow BR itself act as an authenticated HTTP(S) server to simplify deployment.
+- Recommended Skills: Go, Rust, HTTP communication, TLS handling.
+- Mentor(s): kennytm
```
Comment (Soline324) [inline talent-challenge-program/project-ideas.md:23]:
```suggestion
- Mentor(s): [kennytm](https://github.com/kennytm), [PingCAP](https://pingcap.com/en/)
```

**Pull request merged**